app/controllers/candidate_interface/sign_in_controller.rb

```diff
@@ -12,38 +12,47 @@ def create
       SignInCandidate.new(candidate_params[:email_address], self).call
     end
 
+    # This is where users land after clicking the magic link in their email. Because
+    # some email clients preload links in emails, this is an extra step where
+    # they click a button to confirm the sign in.
     def confirm_authentication
-      candidate = retrieve_candidate_by_token || retrieve_candidate_by_user_id
+      authentication_token = AuthenticationToken.find_by_hashed_token(
+        user_type: 'Candidate',
+        raw_token: params[:token],
+      )
 
-      redirect_to(action: :new) and return if candidate.nil?
-
-      if FindCandidateByToken.token_not_expired?(candidate)
+      if authentication_token && authentication_token.still_valid?
         render 'confirm_authentication'
-      else
+      elsif params[:u]
```

Comment:
Not necessarily for this PR but might make code more readable if `:u` was renamed `:user_id` ?

Reply:
Once this is merged we can remove all the user id related code as we will have associated the token with the user